tests/lib/rules/prefer-expect-query-by.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/prefer-expect-query-by');
+
+const ruleTester = new RuleTester({
+  parserOptions: { ecmaVersion: 2015, sourceType: 'module' },
+});
+
+ruleTester.run('prefer-expect-query-by', rule, {
+  valid: [
+    { code: "expect(queryByText('Hello')).not.toBeInTheDocument()" },
+    { code: "expect(rendered.queryByText('Hello')).not.toBeInTheDocument()" },
+    { code: "expect(queryAllByText('Hello')).not.toBeInTheDocument()" },
+    {
+      code: "expect(rendered.queryAllByText('Hello')).not.toBeInTheDocument()",
```

Comment:
Sure, I also included matching for `findBy` now. Thanks for the reminder.

Reply:
Oh sorry, I meant the opposite: `findBy` is a valid query because there is not alternative for it though it throws (we discussed this in the original issue, you can check the conversation). So `findBy` queries must be valid and this rule is all about `getBy` vs `queryBy`

Reply:
@Belco90  Shouldn't we check all the variants of the `getBy` queries too (as it's done in `await-async-queries`) for example?

Reply:
Didn't remember this! Yeah, it seems like a good idea to me, can you add these tests variants @emmenko? As @thomlom mentioned, `await-async-queries` is a good example.

Reply:
The implementation just checks for the query name to start with `getBy`. So implicitly all variants are tested. I don't think we need to manually test for all of them.

Reply:
I mean to add tests for all the different testing library queries, not to manually check them one by one in the rule implementation. Even if that's checked with a query name start, all queries involved here need tests :)

Reply:
Sure I can add the tests for the other queries. I personally find it not necessary but if you prefer so it's fine by me.

Reply:
Cool. As we mentioned, you can replicate the mechanism in `await-async-queries` to automate that

Reply:
This is how I did it: https://github.com/Belco90/eslint-plugin-testing-library/pull/22/commits/53a950c8716b6c63a37646d6cedb7c2a7929586c

Is that ok?